js/background.js

```diff
@@ -295,6 +309,12 @@
 
     function onMessageReceived(ev) {
         var data = ev.data;
+        if (data.message.flags & textsecure.protobuf.DataMessage.Flags.PROFILE_KEY_UPDATE) {
+            var profileKey = data.message.profileKey.toArrayBuffer();
+            return ConversationController.getOrCreateAndWait(data.source, 'private').then(function(sender) {
+              return sender.setProfileKey(profileKey).then(ev.confirm);
```

Comment:
Calling `ev.confirm()` at the end of this save is probably too aggressive. I think we want to wait until all message processing is complete.

Reply:
I think we should avoid initializing a Whisper.Message that we don't want to save. Neither ios or android are saving message records for profile key updates.

Reply:
Also, handleDataMessage will have side effects like bumping the conversation to the top of the list, triggering a notification, etc... which would be weird when no new message bubble is being added.

Reply:
Ah, again I'm mistaken about how flags are used. The design of the data structure suggests that any message can have multiple flags, and be a normal message as well. But you're telling me that a `DataMessage` is one of: 1) a normal message 2) a empty message with any _single_ flag